backend/sftp/sftp.go

```diff
@@ -507,7 +512,39 @@ func NewFs(name, root string, m configmap.Mapper) (fs.Fs, error) {
 		if err != nil {
 			return nil, errors.Wrap(err, "failed to parse private key file")
 		}
-		sshConfig.Auth = append(sshConfig.Auth, ssh.PublicKeys(signer))
+
+		// If a public key has been specified then use that
+		if pubkeyFile != "" {
+			certfile, err := ioutil.ReadFile(pubkeyFile)
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to read cert file")
+			}
+
+			pk, _, _, _, err := ssh.ParseAuthorizedKey(certfile)
+			if err != nil {
+				return nil, errors.Wrap(err, "unable to parse cert file")
+			}
+
+			// And the signer for this, which includes the private key signer
+			// This is what we'll pass to the ssh client.
+			// Normally the ssh client will use the public key built
+			// into the private key, but we need to tell it to use the user
+			// specified public key cert.  This signer is specific to the
+			// cert and will include the private key signer.  Now ssh
+			// knows everything it needs.
+			switch pk.(type) {
+			case *ssh.Certificate: // This is the right type
+			default:
+				return nil, errors.New("Public key file is not a certificate file: " + pubkeyFile)
```

Comment:
picky: the [Go style standard](https://github.com/golang/go/wiki/CodeReviewComments#error-strings) states

> Error strings should not be capitalized (unless beginning with proper nouns or acronyms) 

This is so that the thing which does the final logging does `log.Printf("Failed doing thing: %v", err)` and we only get one capital.

Reply:
OK; I'm still a newbie at GoLang; I wasn't aware of that :-)  Changed!